### PR TITLE
support for Qt 6 builds

### PR DESCRIPTION
**Pull request by 3nids**
This makes the plugin compatible with Qt6 builds.

QGIS minimum version is raised to 3.34, the config dialogs is shown only for 3.40+.

Another approach would be to set the new minimum version to 3.40 (so we would leave the old version for 3.34-3.38), no strong opinion here.

### CODE REVIEW

**Comment by 3nids** [comment]
(The test is failing because the new API from recent commits has not yet reached the docker image)

File: swiss_locator/core/settings.py [hunk 1]
```diff
@@ -24,82 +24,96 @@
...
 from swiss_locator.core.filters.filter_type import FilterType
 
-pluginName = "swiss_locator_plugin"
+PLUGIN_NAME = "swiss_locator_plugin"
 
 
-class Settings(SettingManager):
-    def __init__(self):
-        SettingManager.__init__(self, pluginName)
+class Settings(object):
```
Comment (gacarrillor) [inline swiss_locator/core/settings.py:41]: It seems that `object`is not necessary for Python 3. It was a Python 2 thing.

**Review by gacarrillor — APPROVED** [review]
There are some things going on that I cannot completely grasp (not enough experience handling settings from QGIS), but in general it looks good to me. 👍🏽 

I'll test in GUI and let you know if I find anything strange.

**Comment by gacarrillor** [comment]
> Another approach would be to set the new minimum version to 3.40 (so we would leave the old version for 3.34-3.38), no strong opinion here.

This one looks a bit better to me, but I guess there are pros and cons for both options.
In February 2025, 3.40 will be LTR anyways.

**Comment by gacarrillor** [comment]
For reference, followup https://github.com/opengisch/qgis-swiss-locator/pull/92

**Review by gacarrillor — COMMENTED** [review]
I just stumbled upon this when I click on the locator widget:
image


There seems to be too much space between the two columns in tab WMTS (the space is different with respect to the other 3 tabs):
image

**Comment by 3nids** [comment]
Thanks a lot for the review.

There's one scenario I'm a bit worried about raising to 3.40.
Let's say someone has both versions used with the same profile: you upgrade Swiss Locator when you open 3.40, it becomes unusable when you open 3.34.

But, this might be better indeed than not having access to settings in current LTR.

**Comment by 3nids** [comment]
let's bump to 3.40 and we can revert to 3.34 if needed later on!

**Comment by gacarrillor** [comment]
> Thanks a lot for the review.
> 
> There's one scenario I'm a bit worried about raising to 3.40. Let's say someone has both versions used with the same profile: you upgrade Swiss Locator when you open 3.40, it becomes unusable when you open 3.34.
> 
> Another point to take into account is how important it is to push new releases to 3.34:
> 
>     * with the min 3.34 approach, it's included
> 
>     * with the min 3.40 approach, we need to have a separate branch

Yeah, good points. You know more about pros and cons with respect to the user base, let's hope it works with 3.40 as min and if not we need to be ready to revert as you said. 👍🏽

**Pull request merged**